specs/latest/2.0/index.html

```diff
@@ -1833,24 +1833,30 @@ <h2 class="no-toc">Table of contents</h2>
         <p class="idl-code">
           void compressedTexImage2D(GLenum target, GLint level, GLenum internalformat,
                                     GLsizei width, GLsizei height, GLint border,
-                                    ArrayBufferView srcData, optional GLuint srcOffset = 0)
+                                    ArrayBufferView srcData,
+                                    optional GLuint srcOffset = 0,
+                                    optional GLuint srcLengthOverride = 0)
           <span class="gl-spec">
             (<a href="http://www.khronos.org/registry/gles/specs/3.0/es_spec_3.0.4.pdf#nameddest=section-3.8.6">OpenGL ES 3.0.4 &sect;3.8.6</a>,
             <a href="http://www.khronos.org/opengles/sdk/docs/man3/html/glCompressedTexImage2D.xhtml" class="nonnormative">man page</a>)
           </span>
         </p>
       <dd>
         <p>Reading from <code>srcData</code> begins <code>srcOffset</code> elements into
-           <code>srcData</code>. (Elements are bytes for Uint8Array, int32s for Int32Array, etc.)
-        <p>If there's not enough data in <code>srcData</code> starting at <code>srcOffset</code>,
-           generate INVALID_OPERATION.
+           <code>srcData</code>. (Elements are bytes for Uint8Array, int32s for Int32Array, etc.)</p>
+        <p><code>srcLengthOverride</code> defaults to <code>srcData.length - srcOffset</code>.</p>
```

Comment:
We probably should define the case where srcOffset > srcData.length first? So the srcData.length - srcOffset later will be non-negative.